Tests/LibC/TestFenv.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2023, kleines Filmröllchen <filmroellchen@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibTest/TestCase.h>
+#include <fenv.h>
+
+// TODO: Add tests for floating-point exception management.
+
+// This error happens even with float literals; e.g. -1.099999f becomes -1.099998f.
+// However, it is enough to distinguish the different rounding errors we test for.
+static constexpr double acceptable_float_error = 0.000001;
+
+TEST_CASE(float_round_up)
+{
+    fesetround(FE_UPWARD);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.2f, 0.3f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.3f, 0.4f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.4f, 0.5f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(-1.f + -0.1f, -1.099999f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(1.f + 0.1f, 1.1f, acceptable_float_error);
+}
+
+TEST_CASE(float_round_down)
+{
+    fesetround(FE_DOWNWARD);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.2f, 0.299999f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.3f, 0.4f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.4f, 0.5f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(-1.f + -0.1f, -1.1f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(1.f + 0.1f, 1.099999f, acceptable_float_error);
+}
+
+TEST_CASE(float_round_to_zero)
+{
+    fesetround(FE_TOWARDZERO);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.2f, 0.299999f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.3f, 0.4f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.4f, 0.5f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(-1.f + -0.1f, -1.099999f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(1.f + 0.1f, 1.099999f, acceptable_float_error);
+}
+
+TEST_CASE(float_round_to_nearest)
+{
+    fesetround(FE_TONEAREST);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.2f, 0.3, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.3f, 0.4, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(0.1f + 0.4f, 0.5, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(-1.f + -0.1f, -1.1f, acceptable_float_error);
+    EXPECT_APPROXIMATE_WITH_ERROR(1.f + 0.1f, 1.1f, acceptable_float_error);
+}
+
+// FIXME: Figure out some tests to distinguish nearest from max-magnitude on supported platforms.
```

Comment:
It is not hard in the slightest, if I correctly understood what FE_TOMAXMAGNITUDE is. You just need to start with an "even" float and add half the distance to the next one. For example, `1.0f + 5.9604644775390625e-8f` should give `1.0f` with `FE_TONEAREST` and `1.00000011920928955078f` with `FE_TOMAXMAGNITUDE`.